enhancements/baremetal-config.md

```diff
@@ -0,0 +1,153 @@
+  - "@squeed"
+  - "@abhinavdahiya"
+
+approvers:
+  - "@deads2k"
+
+creation-date: 2019-10-23
+last-updated: 2019-10-23
+status: implemented
+see-also:
+  - "https://github.com/openshift/api/pull/480"  
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+None.
+
+## Summary
+
+The configuration required to bring up a Baremetal IPI (metal3) cluster needs
+to be availabe in a Config Resource. The installer aleady adds some configuration
+to the Infrastructure Config CR. It is also aware of the additional config
+required during the deployment of a metal3 cluster. This enhancement proposes
+to extend the BareMetaPlatformStatus CR to hold these additional values needed
+for a metal3 deployment.
+
+## Motivation
+
+Adding these additional config items to the BareMetalPlatformStatus CR, allows
+versioning of these config items.
+
+### Goals
+
+The goal of this enhancement is to enable the installer to add vital config
+values for a Baremetal IPI deployment to a CR. The MAO will then be able to
+access these values when it kicks off a metal3 cluster.
+
+Update lifecycle for bootimage is an unsolved problem for all platforms. For
+more information, refer to : https://github.com/openshift/os/issues/381.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
+metal3 cluster is up. Handling changes to this value after install time has
+not been explored and that is a non-goal at this time.
+
+### Proposal
+
+To support configuration of a Baremetal IPI (metal3) deployment, a set of new
+config items are required to be made availabe via a Config Resource. The
+BareMetalPlatformStatus is an ideal fit because it contains configuration
+that is specific to a Baremetal deployment.
+
+This proposal aims to add the following to BareMetalPlatformStatus:
+
+1. ProvisioningInterface : This is the interface name on the underlying
+baremetal server which would be physically connected to the provisioning
+network.
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on
+the baremetal server for provisioning. This value should not be in the
+DHCP range and should not be used in the provisioning network for any
+other purpose (should be a free IP address.) It is expected to be provided
+in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its
+CIDR. The ProvisioningIP and the ProvisioningDHCPRange are expected to
+be within this network. The value for this config is expected to be
+provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which
```

Comment:
Is there a reason we shouldn't just use the MachineCIDR?

Reply:
This is a completely different network to the MachineCIDR, it's used for baremetal deployment to pxe boot the servers and access their BMCs via IPMI.

In the installer there is a validation to ensure this network does not overlap with MachineCIDR ref https://github.com/openshift/installer/pull/2449

Reply:
Right, for baremetal we assume at least 2 networks. The MachineCIDR describes the usual network where control plane and application traffic go. This new ProvisioningCIDR describes the network where provisioning traffic goes. That traffic needs to be isolated from the control plane and applications because it involves protocols like DHCP and PXE, and you don't want to mix those with publicly facing traffic, so they are typically on different L2 network segments.

Reply:
Gotcha, Fun.